src/aiida/storage/psql_dos/orm/querybuilder/main.py

```diff
@@ -385,7 +428,7 @@ def build_filters(self, alias: AliasedClass, filter_spec: Dict[str, Any]) -> Opt
                 column_name = path_spec.split('.')[0]
 
                 attr_key = path_spec.split('.')[1:]
-                is_jsonb = bool(attr_key) or column_name in ('attributes', 'extras')
+                is_jsonb = bool(attr_key) or column_name in ('dict', 'attributes', 'extras')
```

Comment:
Had to do this, as the mapping of `dict -> attributes` does not happen until the `try/except` block in the following lines, but the `except` section makes use of `is_jsonb`. It was this or code duplication. Improve if possible.